Shared.Tests/CloudinaryTest.cs

```diff
@@ -2770,56 +2833,91 @@ public void TestClearAllTags()
             });
 
             Assert.Null(getResResult.Tags);
-
-            DelResResult delResult = m_cloudinary.DeleteResources(new DelResParams()
-            {
-                PublicIds = pIds
-            });
-
         }
 
         [Test]
         public void TestAddContext()
         {
+            var publicId = $"{m_apiTest}_TestAddContext";
+
             var uploadParams = new ImageUploadParams()
             {
                 File = new FileDescription(m_testImagePath), 
-                PublicId = "TestContext",
+                PublicId = publicId,
                 Overwrite = true,
                 Type = "upload",
-
+                Tags = m_apiTag
             };
 
-            var uploadResult = m_cloudinary.Upload(uploadParams);
+            m_cloudinary.Upload(uploadParams);
 
             List<string> pIds = new List<string>();
-            pIds.Add("TestContext");
+            pIds.Add(publicId);
 
             ContextResult contextResult = m_cloudinary.Context(new ContextParams()
             {
                 Command = ContextCommand.Add,
                 PublicIds = pIds,
                 Type = "upload",
                 Context = "TestContext"
-
             });
 
             Assert.True(contextResult.PublicIds.Length > 0);
 
-            var getResResult = m_cloudinary.GetResource(new GetResourceParams(pIds[0])
+            m_cloudinary.GetResource(new GetResourceParams(pIds[0])
             {
                 PublicId = pIds[0],
                 Type = "upload",
                 ResourceType = ResourceType.Image
             });
+        }
 
+        [OneTimeTearDown]
+        public override void Cleanup()
+        {
+            base.Cleanup();
+            m_cloudinary.DeleteTransform(m_updateTransformation.ToString());
 
-            DelResResult delResult = m_cloudinary.DeleteResources(new DelResParams()
+            var transforms = new List<object>
             {
-                PublicIds = pIds
-            });
+                m_explicitTransformation,
+                m_updateTransformation,
+                m_apiTest,
+                m_apiTag,
+                m_apiTest1,
+                m_apiTest2, 
+            };
 
-        }
+            transforms.ForEach(t => m_cloudinary.DeleteTransform(t.ToString()));
+
+            var presets = new[] {
+                $"{m_apiTest1}_upload_preset",
+                $"{m_apiTest2}_upload_preset",
+                $"{m_apiTest2}_upload_preset_unsigned",
+                $"{m_apiTest1}_api_test_upload_preset",
+                $"{m_apiTest2}_api_test_upload_preset"
+            };
+            foreach (var preset in presets)
+            {
+                m_cloudinary.DeleteUploadPreset(preset);
+            }
 
+            var resources = new Dictionary<string, string>
```

Comment:
use variables/constants

we can easily get out-of-sync, if someone changes unit test and forgets to update Cleanup()